paasta_tools/cli/cmds/itest.py

```diff
@@ -69,41 +69,52 @@ def paasta_itest(args):
     """Build and test a docker image"""
     service = args.service
     soa_dir = args.soa_dir
+
     if service and service.startswith("services-"):
         service = service.split("services-", 1)[1]
+
     validate_service_name(service, soa_dir=soa_dir)
 
-    tag = build_docker_tag(service, args.commit)
-    run_env = os.environ.copy()
-    run_env["DOCKER_TAG"] = tag
-    cmd = "make itest"
-    loglines = []
+    tags = build_docker_tags(service, args.commit)
 
-    _log(
-        service=service,
-        line="starting itest for %s." % args.commit,
-        component="build",
-        level="event",
-    )
-    returncode, output = _run(
-        cmd,
-        env=run_env,
-        timeout=args.timeout,
-        log=True,
-        component="build",
-        service=service,
-        loglevel="debug",
-    )
-    if returncode != 0:
-        loglines.append("ERROR: itest failed for %s." % args.commit)
-        output = get_jenkins_build_output_url()
-        if output:
-            loglines.append("See output: %s" % output)
-    else:
-        loglines.append("itest passed for %s." % args.commit)
-        if not check_docker_image(service, args.commit):
-            loglines.append("ERROR: itest has not created %s" % tag)
-            returncode = 1
-    for logline in loglines:
-        _log(service=service, line=logline, component="build", level="event")
-    return returncode
+    resultingcode = 0
+
+    for tag in tags:
+        run_env = os.environ.copy()
+        run_env["DOCKER_TAG"] = tag
+        cmd = "make itest"
+        loglines = []
+
+        _log(
+            service=service,
+            line="starting itest for %s." % args.commit,
+            component="build",
+            level="event",
+        )
+
+        returncode, output = _run(
```

Comment:
I don't think you should be running itest multiple times; instead, can we run it once to produce the image, and then tag it multiple times? these images take a while to build

Reply:
That's correct. Found an easy solution here: https://stackoverflow.com/a/35565384